src/hera/workflows/_runner/util.py

```diff
@@ -190,8 +190,10 @@ def _map_function_annotations(function: Callable, template_inputs: Dict[str, str
                     function_kwargs[func_param_name] = load_param_input(
                         param_value, func_param.annotation, param_or_artifact.loader
                     )
-            else:
-                function_kwargs[func_param_name] = get_annotated_artifact_value(param_or_artifact)
+            elif isinstance(param_or_artifact, Artifact):
```

Comment:
nit: why did this change? We already know it's an Artifact here based on what `get_workflow_annotation` can return -- if we want to defend against the case it might not be, it would be clearer to make this an `assert(isinstance(param_or_artifact, Artifact))` in an unconditional `else`; failing that, there should be an else/raise branch.

Reply:
No functional reason for this change, it was mainly for readability while I was working on the feature. I've changed it back for now as I think I managed to avoid too much indentation through some refactoring.